src/drivers/adc/ADC.cpp

```diff
@@ -148,9 +149,28 @@ void ADC::update_adc_report(hrt_abstime now)
 
 	for (unsigned i = 0; i < max_num; i++) {
 		adc.channel_id[i] = _samples[i].am_channel;
-		adc.channel_value[i] = _samples[i].am_data * 3.3f / px4_arch_adc_dn_fullcount();
+		adc.raw_data[i] = _samples[i].am_data;
+		adc.v_ref[i] = 3.3f;	// TODO: fill the true v_ref, maybe new macros for embed device?
+#ifdef ADC_BATTERY_VOLTAGE_CHANNEL
```

Comment:
@SalimTerryLi - why is this needed? We want to remove the BOARD_ADC_POS_REF_V_FOR_VOLTAGE_CHAN and BOARD_ADC_POS_REF_V_FOR_CURRENT_CHAN  and take the ref from the ADC via the new API

Reply:
The new PR has not been merged yet. I'm waiting for that to provide such a API....

Reply:
@SalimTerryLi - it is merged now

Reply:
Done with 86808ed